File: modules/lindstrom-hsm-basetypes/ios/LindstromHsmBasetypesModule.swift
import ExpoModulesCore
import HsMBaseTypes

public class LindstromHsmBasetypesModule: Module {
    public func definition() -> ModuleDefinition {
        Name("LindstromHsmBasetypes")
       
        Function("parseCodeWithResult") { (code: String) in
            let result = HsMDeviceCodeParser.shared.parseCodeWithResult(qrCodeString: code)
           
            return [
                "success": result.success,
                "qrCode": result.qrCode.debugDescription // This returns "Optional(HsMDeviceCodeParser.QrCodeResult)" which is not the string we want, but we do not need it in the app.
            ]
        }
    }
}

File: modules/lindstrom-hsm-basetypes/android/src/main/java/expo/modules/lindstromhsmbasetypes/LindstromHsmBasetypesModule.kt
package expo.modules.lindstromhsmbasetypes

import com.hagleitner.basetypes.qr.HsMDeviceCodeParser
import com.hagleitner.basetypes.qr.exception.qr.InvalidQrCodeFormat
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import java.net.URL

class LindstromHsmBasetypesModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("LindstromHsmBasetypes")

    Function("parseCodeWithResult") { code: String ->
      val result = HsMDeviceCodeParser.parseCodeWithResult(code)
      mapOf(
        "success" to result.success,
        "qrCode" to result.qrCode?.toString()
      )
    }
  }
}
